lib/puma/plugin/systemd.rb

```diff
@@ -29,31 +29,6 @@
   #   https://www.freedesktop.org/software/systemd/man/sd-daemon.html
   #
   class Systemd
-    # Do we have a systemctl binary? This is a good indicator whether systemd
-    # is installed at all.
-    def present?
-      ENV["PATH"].split(":").any? { |dir| File.exists?(File.join(dir, "systemctl")) }
-    end
-
-    # Is the system currently booted with systemd?
-    #
-    # We could check for the systemd run directory directly, but we can't be
-    # absolutely sure of it's location and breaks encapsulation. We can be sure
-    # that systemctl is present on a systemd system and understand whether
-    # systemd is running. The systemd-notify binary actually recommends this.
-    #
-    #   An alternate way to check for this state is to call systemctl(1) with
-    #   the is-system-running command. It will return "offline" if the system
-    #   was not booted with systemd.
-    #
-    # See also sd_booted:
-    #
-    #   https://www.freedesktop.org/software/systemd/man/sd_booted.html
-    #
-    def booted?
-      IO.popen(["systemctl", "is-system-running"], &:read).chomp != "offline"
```

Comment:
From that page:
> Internally, this function checks whether the directory /run/systemd/system/ exists. A simple check like this can also be implemented trivially in shell or any other language.

Perhaps this better? It doesn't have to exec anything which is safer.

Reply:
I tested this and it worked. I know the original comment seemed to warn against this:

```
    # We could check for the systemd run directory directly, but we can't be
    # absolutely sure of it's location and breaks encapsulation. We can be sure
    # that systemctl is present on a systemd system and understand whether
    # systemd is running. The systemd-notify binary actually recommends this.
```

However, if that page on sd_booted is correct this ought to be a safe option that doesn't make things more complex for SELinux users.

Reply:
I can't find that part in the current man pages. Perhaps the recommendations have changed?

Reply:
Right, I was considering the edge case that systemd might be installed with a different prefix and so best deferred to the system's `$PATH` a la `pkg-config`. But a non-default prefix would be _very_ irregular. I also thought that the implementation of sd_booted was more complex, but it [is not](https://github.com/systemd/systemd/blob/master/src/libsystemd/sd-daemon/sd-daemon.c#L622-L634), and is in fact just detecting that one and only directory path. So I'm not even sure a non-default directory is _possible_.

So this implementation looks good. :+1: